README.md

```diff
@@ -1,47 +1,185 @@
+* Open the `app-boilerplate` folder with VSCode.
+* Open the vscode tasks with  `ctrl + shift + b` (`command + shift + p` on a Mac) and run the following actions :
+    * Pull and run the [ledger-app-dev-tools]() docker image by selecting `Run dev-tools image` (select the task for your platform).
+    * Build the for the device model of your choice with `Build app`.
+    * Test your binary on [Speculos](https://github.com/LedgerHQ/speculos) with `Run Speculos`.
+
+:information_source: The VsCode tasks also allow you to perform the following actions :
+
+* Install functional tests Python requirements (mandatory before executing functional tests),
+* [Execute functional tests](#test),
+* Open a terminal in your docker container,
+* Clean build files,
+* Load the app on a physical device.
+
+You can find all the available tasks definitions in `.vscode/tasks.json`.
+
+### With a shell and the docker builder image
 
 The app-builder docker image [from this repository](https://github.com/LedgerHQ/ledger-app-builder) contains all needed tools and library to build and load an application.
 You can download it from the ghcr.io docker repository:
 
 ```shell
-sudo docker pull ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder-full
+sudo docker pull ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder
 ```
 
 You can then enter this development environment by executing the following command from the directory of the application `git` repository:
 
+**Linux (Ubuntu)**
+
 ```shell
-sudo docker run --rm -ti --user "$(id -u)":"$(id -g)" -v "$(realpath .):/app" ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder-full
+sudo docker run --rm -ti --user "$(id -u):$(id -g)" --privileged -v "/dev/bus/usb:/dev/bus/usb" -v "$(realpath .):/app" ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder
+```
+
+**macOS**
+
+```shell
+sudo docker run  --rm -ti --user "$(id -u):$(id -g)" --privileged -v "$(pwd -P):/app" ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder
+```
+
+**Windows (with PowerShell)**
+
+```shell
+docker run --rm -ti --privileged -v "$(Get-Location):/app" ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
 ```
 
 The application's code will be available from inside the docker container, you can proceed to the following compilation steps to build your app.
 
-### Without the docker image builder
+## Compilation and load
 
-Be sure to have your environment correctly set up (see [Getting Started](https://developers.ledger.com/docs/nano-app/introduction/)) and [ledgerblue](https://pypi.org/project/ledgerblue/) and installed.
+### Compilation
 
-If you want to benefit from [vscode](https://code.visualstudio.com/) integration, it's recommended to move the toolchain in `/opt` and set `BOLOS_ENV` environment variable as follows
+Once you have setup a working development environment (by following the [shell with docker](#with-a-shell-and-the-docker-builder-image) approach), you can build and load the app.
```

Comment:
perhaps we could recommend to use the vscode compilation action here as you did for test section

Reply:
Done.